app/models/manageiq/providers/openshift/container_manager/metrics_collector_worker.rb

```diff
@@ -7,5 +7,14 @@ class Openshift::ContainerManager::MetricsCollectorWorker < BaseManager::Metrics
     def friendly_name
       @friendly_name ||= "C&U Metrics Collector for OpenShift"
     end
+
+    # Override PerEmsTypeWorkerMixin.all_valid_ems_in_zone to limit metrics collection
+    def self.all_ems_in_zone
+      super.select do |ems|
+        ems.supports?(:metrics).tap do |supported|
+          _log.info("Skipping [#{ems.name}] since it has no metrics endpoint") unless supported
+        end
+      end
+    end
```

Comment:
NOTE I wanted to subclass this using `ActsAsStiLeafClass` but that messes up the settings path.  It would be great if we could skip classes which act as an STI leaf class when building the settings path so that we don't need to nest settings under the parent class like: https://github.com/ManageIQ/manageiq-providers-ibm_power_vc/blob/master/config/settings.yml#L133-L134

Reply:
That's interesting - I still want to get my flatten worker settings PR over the finish line anyway which I think would make this a moot point